node/reqres.js

```diff
@@ -134,19 +134,29 @@ function TChannelOutgoingResponse(id, options, sendFrame) {
 
 inherits(TChannelOutgoingResponse, EventEmitter);
 
-TChannelOutgoingResponse.prototype.send = function send(err, res1, res2) {
+TChannelOutgoingResponse.prototype.sendOk = function send(res1, res2) {
     var self = this;
     if (self.sent) {
         throw new Error('response already sent');
     }
+
     self.sent = true;
-    if (err) {
-        self.ok = false;
-        var errArg = isError(err) ? err.message : JSON.stringify(err); // TODO: better
-        self.sendFrame(self.arg1, res1, errArg);
-    } else {
-        self.sendFrame(self.arg1, res1, res2);
+    self.ok = true;
+
+    self.sendFrame(self.arg1, res1, res2);
+    self.emit('end');
+};
+
+TChannelOutgoingResponse.prototype.sendNotOk = function sendNotOk(res1, res2) {
```

Comment:
hmm.... it'd be nice if we could pass in application error codes through this... it's poorly / not spec'd, but applications could benefit form defining an error code space >0.